File: server/CinemasService/Models/Layout.cs
using System.ComponentModel.DataAnnotations;

namespace CinemasService.Models
{
    public class Layout
    {
        public Guid Id { get; set; }

        [MaxLength(64)]
        public required string Name { get; set; }
        public required DateTime CreatedAt { get; set; }

        public required int Rows{ get; set; }
        public required int Columns { get; set; }

        public DateTime? DeletedAt { get; set; }
        public ICollection<LayoutElement> Elements { get; set; } = new List<LayoutElement>();

        // Totales, para evitar calcular a menudo
        public required int StandardSeats { get; set; }
        public required int VipSeats { get; set; }
        public required int AccesibleSeats { get; set; }
        public required int DisabledSeats { get; set; }
    }
}

File: ui/cms/src/model/layout/Layout.ts
export enum LayoutElementType {
  STANDARD = 0,
  VIP = 1,
  DISABLED = 2, // Disabled meaning its not available
  ACCESSIBLE = 3,
  BLANK = 4,
  AISLE = 5,
}

export interface LayoutElement {
  id: string;
  label: string;
  index: number;
  type: LayoutElementType;
}

export interface Layout {
  id: string;
  name: string;
  rows: number;
  columns: number;
  createdAt: Date;
  elements: LayoutElement[];
  standardSeats: number;
  vipSeats: number;
  accesibleSeats: number;
  disabledSeats: number;
}

export interface LayoutSummary {
  id: string;
  name: string;
  rows: number;
  columns: number;
  createdAt: Date;
  standardSeats: number;
  vipSeats: number;
  accesibleSeats: number;
  disabledSeats: number;
}


export interface LayoutSearchFilters {
  name?: string;
}

export interface LayoutElementCreateDto {
  label: string;
  index: number;
  type: LayoutElementType;
}

export interface LayoutCreateDto {
  name: string;
  elements: LayoutElementCreateDto[];
  rows: number;
  columns: number;
}
